### ISSUE TITLE
Logits shift in loss computation

### ISSUE THREAD

**Issue by shivamag125** [jzhang38/EasyContext#39]
While the computing the loss L136, shouldn't the logits and targets be rolled to account for next token prediction? 

Similar to https://github.com/huggingface/transformers/blob/main/src/transformers/models/llama/modeling_llama.py#L1092

**Comment by shivamag125** [comment]
Edit- I see that you took care of it while preparing the targets.
